google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GrpcSpannerRpc.java

```diff
@@ -325,14 +326,18 @@ public Operation getOperation(String name) throws SpannerException {
   }
 
   @Override
-  public com.google.spanner.v1.Session createSession(
-      String databaseName, @Nullable Map<Option, ?> options) {
-    CreateSessionRequest request =
-        CreateSessionRequest.newBuilder().setDatabase(databaseName).build();
+  public Session createSession(
+      String databaseName, @Nullable Map<String, String> labels, @Nullable Map<Option, ?> options) {
+    CreateSessionRequest.Builder request =
+        CreateSessionRequest.newBuilder().setDatabase(databaseName);
+    if (labels != null && !labels.isEmpty()) {
+    	  Session.Builder session = Session.newBuilder().putAllLabels(labels);
```

Comment:
I'm missing context here, but this seems weird. In `creteSession()` method, in case if there are any labels provided you _create_ some additional session on client side, to use it as an input parameter for server call, which will return the actual session? In general, why is `session` a field in `CreateSessionRequest` object?

Reply:
We are not creating additional session. CreateSession rpc would create exactly 1 session irrespective of whether the session field is set or not. If there are no labels, we do not need to set the session field. 
I think session being a field in CreateSessionRequest is like any other Create* rpc.
https://cloud.google.com/apis/design/standard_methods#create